### PR TITLE
Add disabling footstep sounds and parallax to the lag switch panel

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
At the last Toolbox Tournament, @MrStonedOne pointed out some big performance consumers. Some are disablable through easy means, but others were not. This adds two easy ones to the mix.

## Changelog

:cl:
admin: Added options to disable footstep sounds and parallax to the lag switch panel.
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
trips btw

File: code/datums/elements/footstep.dm [hunk 1]
```diff
@@ -92,6 +94,9 @@
 /datum/element/footstep/proc/play_simplestep(mob/living/source)
 	SIGNAL_HANDLER
 
+	if (SHOULD_DISABLE_FOOTSTEPS(source))
```
Comment (optimumtact) [inline code/datums/elements/footstep.dm:97]: isn't this kind of lazy in that it still requires a proc call?

really teh element should get beleted and the signals deregged

File: code/controllers/subsystem/lag_switch.dm [hunk 2]
```diff
@@ -111,6 +111,19 @@ SUBSYSTEM_DEF(lag_switch)
...
 		if(DISABLE_NON_OBSJOBS)
 			world.update_status()
+		if(DISABLE_PARALLAX)
+			if (state)
+				to_chat(world, span_boldannounce("Parallax has been disabled for performance concerns."))
+			else
+				to_chat(world, span_boldannounce("Parallax has been re-enabled."))
+
+			for (var/mob/mob as anything in GLOB.mob_list)
+				mob.hud_used?.update_parallax_pref()
```
Comment (optimumtact) [inline code/controllers/subsystem/lag_switch.dm:121]: does this remove the relevant background planes from the clients view?

File: code/datums/elements/footstep.dm (hunk 1 above)
Comment (Mothblocks) [reply code/datums/elements/footstep.dm:97]: I asked MSO about this as well, here's my collected thoughts:

The big performance loss was not in the fact that this is being called every Moved, but the playing of sounds specifically.

We want to be able to easily toggle footsteps on/off, but we also want people with TRAIT_EXEMPT_WHATEVER_IT_WAS to still have footsteps.

This is the easiest way to do all that cleanly, and solves the big performance loss.

File: code/controllers/subsystem/lag_switch.dm (hunk 2 above)
Comment (Mothblocks) [reply code/controllers/subsystem/lag_switch.dm:121]: https://user-images.githubusercontent.com/35135081/142786193-ea690fe7-0ef6-4633-b1c5-c09d7587e61a.mp4

Does this answer your question, or am I mistaken?

File: code/datums/elements/footstep.dm (hunk 1 above)
Comment (optimumtact) [reply code/datums/elements/footstep.dm:97]: okay i trust you
my child would type knife

File: code/controllers/subsystem/lag_switch.dm (hunk 2 above)
Comment (optimumtact) [reply code/controllers/subsystem/lag_switch.dm:121]: yeah

**Pull request merged**